numba/array_analysis.py

```diff
@@ -79,7 +79,7 @@ def codegen(context, builder, sig, args):
         zero = llvmlite.ir.Constant(idx.type, 0)
         is_negative = builder.icmp_signed('<', rem, zero)
         wrapped_rem = builder.add(rem, size)
-        is_oversize = builder.icmp_signed('>', wrapped_rem, size)
+        is_oversize = builder.icmp_signed('>=', wrapped_rem, size)
```

Comment:
Can the original sample that identified this issue be included as a test please?

Reply:
The test case is based off of test_parfor_slice19 btw.

Reply:
Done.